unwind/src/libunwind_shim.rs

```diff
@@ -120,19 +124,28 @@ pub unsafe extern "C" fn _Unwind_FindEnclosingFunction(pc: *mut c_void) -> *mut
 #[no_mangle]
 pub unsafe extern "C" fn _Unwind_RaiseException(exception: *mut _Unwind_Exception) -> _Unwind_Reason_Code {
     (*exception).private_contptr = None;
-    DwarfUnwinder::default().trace(|frames| unwind_tracer(frames, exception));
+    ::glue::registers(|registers| {
+        if let Some(registers) = unwind_tracer(registers, exception) {
+            ::glue::land(&registers);
+        }
+    });
     unreachable!();
 }
 
-unsafe fn unwind_tracer(frames: &mut ::StackFrames, exception: *mut _Unwind_Exception) {
+// We want to ensure cleanup before returning.
+#[inline(never)]
```

Comment:
Is this actually necessary? I don't think inlining is allowed to change the drop order.

Reply:
It's probably not necessary. I was wanting to ensure `_Unwind_RaiseException` didn't have any cleanup at all, because we have to be sure we don't try to run that cleanup when unwinding. But preventing inlining doesn't actually matter, because even if this is inlined, the cleanup would go into the closure, not `_Unwind_RaiseException`.

Reply:
Got it. Just wanted to make sure you're on the same page. Perhaps consider changing the comment to clarify this or just remove it entirely if it doesn't have a clear purpose.